### ISSUE TITLE
Preserve original data anisotropy?

### ISSUE THREAD

**Issue by dpshepherd** [PreibischLab/BigStitcher-Spark#6]
Hi all,

We've got this working fairly well locally. Still struggling with our school Hadoop cluster, which I think is a config issue.

A lot of our data has anisotropic xy pixel size vs z steps. What is the best way to get the BigStitcher-Spark affine fusion to act the same way as the "preserve original data anisotropy" setting in BigStitcher?

One thought I had was to edit the XML to change the calibration before fusion.

Thanks!

**Comment by StephanPreibisch** [comment]
Hi, that would work. But I also added it for the affine fusion right now with commit https://github.com/PreibischLab/BigStitcher-Spark/commit/e8a66ed892b0d8dbc9ce7fd88a382b60fcf05874

Please let me know if it works. As usual it is self-explanatory from the command line parameters ;)

**Comment by dpshepherd** [comment]
Thanks!!

Running a fusion now and will update once it finishes. The log shows that the anisotropy factor was correctly calculated from the XML file.

**Comment by dpshepherd** [comment]
The fusion ran and the log says that preserve anisotropy is `true `and calculated the anisotropy correctly, but the resulting N5 has the same number of z planes when fused with or without the anisotropy flag activated,.

With `--preserveAnisotropy` = `true`, the N5 should have 2557 z planes. However, both `true `and `false` result in a fused file with 10644 z planes. The anisotropy factor is 1 um /.24 um = 4.16667. This matches 2557 * (1/.24) = 10644.

Attached is the xml.

[c13m34_24_bdv.zip](https://github.com/PreibischLab/BigStitcher-Spark/files/7996430/c13m34_24_bdv.zip)

**Comment by StephanPreibisch** [comment]
Hi, it should be fixed now ... could you please try again (sorry).

**Comment by dpshepherd** [comment]
Running with new changes now. Number of blocks to fuse is smaller in the log, which suggests it is working. Will update again once done.

Thanks!

**Comment by StephanPreibisch** [comment]
Fingers crossed! Do you also need non-rigid

**Comment by dpshepherd** [comment]
Nope, rigid is good for our purposes.

**Comment by dpshepherd** [comment]
Correct number of z planes! But - BDV N5 viewer doesn't find a dataset anymore. I can load it in Python using Zarr/N5 tools, which is how I checked.

And muuuch faster now.

**Comment by StephanPreibisch** [comment]
When I tested it it worked like a charm ... how exactly are you opening it? The only difference is a different size, so not sure what could cause that?

**Comment by dpshepherd** [comment]
BigDataViewer -> N5 Viewer. When I click "detect datasets", it doesn't find the contents of the directory (should be Alexa647/s0).

The previous results opened just fine. And I can open the dataset in Python using Zarr with the N5 backend just fine.

![image](https://user-images.githubusercontent.com/26783318/152458267-12267145-a76c-433a-8ff2-4f76a40a600f.png)

**Comment by StephanPreibisch** [comment]
Can you please also paste a screenshot of the actual directory?

**Comment by StephanPreibisch** [comment]
How does File > Import N5 behave?

**Comment by StephanPreibisch** [comment]
usually I have to go into the Alexa647 directory and then press "detect datasets"

**Comment by dpshepherd** [comment]
> Can you please also paste a screenshot of the actual directory? Ideally with permissions?
```linux
(base) dps@qi2labserver:/mnt/opm2/20220120/n5$ tree -L 3 -p output.n5
output.n5
├── [drwxrwxrwx]  Alexa647
│   └── [drwxrwxrwx]  s0
│       ├── [drwxrwxrwx]  0
│       ├── [drwxrwxrwx]  1
│       ├── [drwxrwxrwx]  10
│       ├── [drwxrwxrwx]  2
│       ├── [drwxrwxrwx]  3
│       ├── [drwxrwxrwx]  4
│       ├── [drwxrwxrwx]  5
│       ├── [drwxrwxrwx]  6
│       ├── [drwxrwxrwx]  7
│       ├── [drwxrwxrwx]  8
│       ├── [drwxrwxrwx]  9
│       └── [-rwxrwxrwx]  attributes.json
└── [-rwxrwxrwx]  attributes.json
```

> How does File > Import N5 behave?

Same behavior, no datasets are detected.

> usually I have to go into the Alexa647 directory and then press "detect datasets"

Same behavior, no datasets are detected.

**Comment by dpshepherd** [comment]
I was able to load the dataset in Python using the [Zarr N5 API](https://zarr.readthedocs.io/en/stable/api/n5.html), run some image processing blocks using our own code, and save the output as a ome.tiff using [tifffile](https://github.com/cgohlke/tifffile).

**Comment by StephanPreibisch** [comment]
could you please also post the contents of both attributes.json? I just want to understand what is going wrong ...

**Comment by dpshepherd** [comment]
```linux
(base) dps@qi2labserver:/mnt/opm2/20220120/n5/output.n5$ more attributes.json
{"n5":"2.2.1"}
```

**Comment by StephanPreibisch** [comment]
and the one in s0?

**Comment by dpshepherd** [comment]
```linux
(base) dps@qi2labserver:/mnt/opm2/20220120/n5/output.n5/Alexa647/s0$ more attributes.json
{"min":[-197,-34119,1579],"compression":{"type":"gzip","useZlib":false,"level":1},"blockSize":[512,512,512],"dataType":"uint16","dimensions":[5479,3665,2557]}
```

**Comment by StephanPreibisch** [comment]
this is all correct, that is so weird ...

**Comment by dpshepherd** [comment]
Let me try a fresh Fiji install.

**Comment by StephanPreibisch** [comment]
can you send me everything except the directories in s0?

**Comment by dpshepherd** [comment]
Yes, give me a couple hours. I need to help the group get their posters finished for BPS meeting!

**Comment by StephanPreibisch** [comment]
So I re-created the directory structure with the two attributes.json and it works fine for m
<img width="1000" alt="Screen Shot 2022-02-04 at 1 35 27 PM" src="https://user-images.githubusercontent.com/1420587/152584256-50fe360a-cbf2-4956-b9c3-5d0fc354d61e.png">
e

**Comment by dpshepherd** [comment]
What version of the n5-viewer.jar do you have? I just downloaded a fresh Fiji, updated it, and it is `n5-viewer_fiji-4.5.0.jar`. On this one, there are more windows for metadata types.

![image](https://user-images.githubusercontent.com/26783318/152720693-3242b879-5fc4-4fa6-b726-0f13d7d68f93.png)

If I don't update Fiji, I can detect a dataset but then get an error

![image](https://user-images.githubusercontent.com/26783318/152721065-2c6a0b47-5f52-4c2c-95fc-37e3c71f4d2c.png)

Here is the directory structure without the data directories.
[output_nos0.n5.zip](https://github.com/PreibischLab/BigStitcher-Spark/files/8012166/output_nos0.n5.zip)

**Comment by tpietzsch** [comment]
Is this still a problem or can we close this issue? Or rather split it, because the original "preserve anisotropy" issue seems to be fixed?
(I'm just reading through the BigStitcher-Spark issues at the moment to get a sense of the type of problems...)

**Comment by dpshepherd** [comment]
From our end, the Fiji loading issue is still open. Splitting makes sense to me, as the data is fused with the correct anisotropy and accessible (for us) via Python.
